x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/query/AbstractStringScriptFieldQuery.java

```diff
@@ -6,63 +6,27 @@
 
 package org.elasticsearch.xpack.runtimefields.query;
 
-import org.apache.lucene.index.LeafReaderContext;
-import org.apache.lucene.search.ConstantScoreScorer;
-import org.apache.lucene.search.ConstantScoreWeight;
-import org.apache.lucene.search.DocIdSetIterator;
-import org.apache.lucene.search.IndexSearcher;
-import org.apache.lucene.search.ScoreMode;
-import org.apache.lucene.search.Scorer;
-import org.apache.lucene.search.TwoPhaseIterator;
-import org.apache.lucene.search.Weight;
 import org.elasticsearch.script.Script;
 import org.elasticsearch.xpack.runtimefields.StringScriptFieldScript;
 
-import java.io.IOException;
 import java.util.List;
-import java.util.Objects;
 
 /**
  * Abstract base class for building queries based on {@link StringScriptFieldScript}.
  */
-abstract class AbstractStringScriptFieldQuery extends AbstractScriptFieldQuery {
-    private final StringScriptFieldScript.LeafFactory leafFactory;
+abstract class AbstractStringScriptFieldQuery extends AbstractScriptFieldQuery<StringScriptFieldScript> {
 
     AbstractStringScriptFieldQuery(Script script, StringScriptFieldScript.LeafFactory leafFactory, String fieldName) {
```

Comment:
Maybe leafFactor should be a function everywhere like it is for long, because the function is all we need after all.

Reply:
I think they'll end up being a `Function<LeafReaderContext, StringRuntimeValues>` or something like that when we go to use them in grok or source only fields. But I'm not really sure to be honest. I figured we'd make the change when we got to one of those.